NDTensors/src/lib/BlockSparseArrays/src/blocksparsearray/blocksparsearray.jl

```diff
@@ -118,3 +118,8 @@ blockstype(::Type{<:BlockSparseArray{<:Any,<:Any,<:Any,B}}) where {B} = B
 ##   # TODO: Preserve GPU data!
 ##   return BlockSparseArray{elt}(undef, axes)
 ## end
+
+function Base.adjoint(A::BlockSparseMatrix)
+  T = Base.promote_op(adjoint, eltype(A))
+  return BlockSparseArray{T,2}(adjoint(A.blocks), reverse(A.axes))
```

Comment:
```suggestion
  return BlockSparseMatrix{T}(adjoint(blocks(A)), dual.(reverse(axes(A))))
```
I'm suggesting using accessor methods rather than accessing fields directly, and also adding a `dual` call to handle graded axes.